src/realtime_metrics/main.py

```diff
@@ -79,6 +76,11 @@ def run_analysis():
     
     for trip_id in trips.keys():
         trip_updates = trips[trip_id]
+        if len(trip_updates) == 0:
+            continue
+        trip_updates.sort(key=lambda u: u[0].timestamp.replace(tzinfo=timezone.utc).timestamp())
+        time_frame_start = int(trip_updates[0][0].timestamp.replace(tzinfo=timezone.utc).timestamp() / 60) * 60
+        time_frame_end = int(trip_updates[-1][0].timestamp.replace(tzinfo=timezone.utc).timestamp() / 60) * 60
```

Comment:
```suggestion
        time_frame_start = int(trip_updates[0][0].timestamp.replace(tzinfo=timezone.utc).timestamp())
        time_frame_end = int(trip_updates[-1][0].timestamp.replace(tzinfo=timezone.utc).timestamp())
```

Reply:
This would remove the rounding down to the last minute. While it is not necessary to have the timeframe already rounded, it make debugging easier, so I would keep it.

Reply:
Did you consider to use https://www.w3schools.com/python/ref_func_round.asp ? This int(x / 60) * 60 is a litlle bit confusing imho.

Reply:
I found a function to drop the seconds on the `datetime` object directly. It should be much more understandable now.